src/sandstorm/fuse.c++

```diff
@@ -866,26 +853,25 @@ inline int64_t toNanos(const struct timespec& ts) {
   return ts.tv_sec * 1000000000ll + ts.tv_nsec;
 }
 
-class FileImpl final: public fuse::File::Server {
+class FileImpl final: public fuse::File, public kj::Refcounted {
 public:
   explicit FileImpl(kj::StringPtr path) {
     int ifd;
     KJ_SYSCALL(ifd = open(path.cStr(), O_RDONLY), path);
     fd = kj::AutoCloseFd(ifd);
   }
 
-protected:
-  kj::Promise<void> read(ReadContext context) override {
-    auto params = context.getParams();
-    auto size = params.getSize();
-    auto offset = params.getOffset();
+  kj::Own<fuse::File> addRef() override {
```

Comment:
Actually, no, forget I said that, that's wrong... `addRef()` exists for the sake of people using the abstract interface. Inheriting Refcounted privately wouldn't work because `kj::refcounted<T>()` wouldn't be able to access the private parent class.